sdl_android/src/main/java/com/smartdevicelink/protocol/WiProProtocol.java

```diff
@@ -231,58 +231,60 @@ else if (protocolMsg.getBulkData() != null) {
 		}
 		
 		synchronized(messageLock) {
-			if (data.length > getMtu(sessionType)) {
+			if (data != null) {
+				if (data.length > getMtu(sessionType)) {
 
-				messageID++;
-	
-				// Assemble first frame.
-				Long mtu = getMtu(sessionType);
-				int frameCount = new Long(data.length / mtu).intValue();
-				if (data.length % mtu > 0) {
-					frameCount++;
-				}
-				//byte[] firstFrameData = new byte[HEADER_SIZE];
-				byte[] firstFrameData = new byte[8];
-				// First four bytes are data size.
-				System.arraycopy(BitConverter.intToByteArray(data.length), 0, firstFrameData, 0, 4);
-				// Second four bytes are frame count.
-				System.arraycopy(BitConverter.intToByteArray(frameCount), 0, firstFrameData, 4, 4);
-
-				SdlPacket firstHeader = SdlPacketFactory.createMultiSendDataFirst(sessionType, sessionID, messageID, getMajorVersionByte(),firstFrameData,protocolMsg.getPayloadProtected());
-				firstHeader.setPriorityCoefficient(1+protocolMsg.priorityCoefficient);
-				//Send the first frame
-				handlePacketToSend(firstHeader);
-				
-				int currentOffset = 0;
-				byte frameSequenceNumber = 0;
-
-				for (int i = 0; i < frameCount; i++) {
-					if (i < (frameCount - 1)) {
-	                     ++frameSequenceNumber;
-	                        if (frameSequenceNumber ==
-	                                SdlPacket.FRAME_INFO_FINAL_CONNESCUTIVE_FRAME) {
-	                            // we can't use 0x00 as frameSequenceNumber, because
-	                            // it's reserved for the last frame
-	                            ++frameSequenceNumber;
-	                        }
-					} else {
-						frameSequenceNumber = SdlPacket.FRAME_INFO_FINAL_CONNESCUTIVE_FRAME;
-					} // end-if
+					messageID++;
+		
+					// Assemble first frame.
+					Long mtu = getMtu(sessionType);
+					int frameCount = new Long(data.length / mtu).intValue();
+					if (data.length % mtu > 0) {
+						frameCount++;
+					}
+					//byte[] firstFrameData = new byte[HEADER_SIZE];
+					byte[] firstFrameData = new byte[8];
+					// First four bytes are data size.
+					System.arraycopy(BitConverter.intToByteArray(data.length), 0, firstFrameData, 0, 4);
+					// Second four bytes are frame count.
+					System.arraycopy(BitConverter.intToByteArray(frameCount), 0, firstFrameData, 4, 4);
+
+					SdlPacket firstHeader = SdlPacketFactory.createMultiSendDataFirst(sessionType, sessionID, messageID, getMajorVersionByte(),firstFrameData,protocolMsg.getPayloadProtected());
+					firstHeader.setPriorityCoefficient(1+protocolMsg.priorityCoefficient);
+					//Send the first frame
+					handlePacketToSend(firstHeader);
 					
-					int bytesToWrite = data.length - currentOffset;
-					if (bytesToWrite > mtu) {
-						bytesToWrite = mtu.intValue();
+					int currentOffset = 0;
+					byte frameSequenceNumber = 0;
+
+					for (int i = 0; i < frameCount; i++) {

```

Comment:
Can you fix the formatting with this for loop? The tabs/spaces seem to be way off in this loop and included conditionals.